File: Aplicacao/backend/FluxoCaixa/Microservicos/FluxoCaixa.Integracoes/Models/ConsolidadoDiario.cs
using Amazon.DynamoDBv2.DataModel;

namespace FluxoCaixa.Integracoes.Models;

[DynamoDBTable("ConsolidadosDiarios")] // Define a tabela no DynamoDB
public class ConsolidadoDiario
{
    [DynamoDBHashKey] // Chave de Partição (Partition Key)
    public string ContaId { get; set; } = string.Empty;

    [DynamoDBRangeKey] // Chave de Ordenação (Sort Key)
    public DateTime Data { get; set; } // Data do consolidado

    [DynamoDBProperty]
    public decimal TotalDebitos { get; set; }

    [DynamoDBProperty]
    public decimal TotalCreditos { get; set; }

    [DynamoDBProperty]
    public decimal Saldo { get; set; }

    public ConsolidadoDiario() {} // Construtor vazio exigido pelo DynamoDB

    public ConsolidadoDiario(string contaId, DateTime data, decimal totalDebitos, decimal totalCreditos, decimal saldo)
    {
        ContaId = contaId;
        Data = data;
        TotalDebitos = totalDebitos;
        TotalCreditos = totalCreditos;
        Saldo = saldo;
    }
}

File: Aplicacao/frontend/desafio-azurebrazil-cloud/src/models/ConsolidadoDiario.js
export class ConsolidadoDiario {
    constructor(
      contaId = "",
      data = new Date().toISOString().split("T")[0],
      totalDebitos = 0.0,
      totalCreditos = 0.0,
      saldo = 0.0
    ) {
      this.contaId = contaId;
      this.data = data;
      this.totalDebitos = totalDebitos;
      this.totalCreditos = totalCreditos;
      this.saldo = saldo;
    }
  }
  